javascript/javascript3/week3/quiz app.js

```diff
@@ -0,0 +1,289 @@
+const tryAgainButton = document.querySelector("#try-again-button");
+const newQuizButton = document.querySelector("#new-quiz-button");
+const player1 = document.querySelector("#player1");
+const player2 = document.querySelector("#player2");
+const timerElement = document.querySelector("#timer");
+const pages = document.querySelectorAll(".page");
+
+
+
+let countdown = 10; 
+let currentQuestion = 0;
+let score = 0;
+let incorrectAnswers = [];
+const playersInfo = [
+  { name: "", score: 0 },
+  { name: "", score: 0 },
+];
+let quizQuestions = [];
+const userQuiz = [];
+const maxScore = 10;
+
+
+
+
+// Timer countdown
+const startTimer = () => {setInterval(() => {
+  console.log(countdown);
+  timerElement.textContent = countdown;  
+  countdown--; 
+
+  if (countdown < 0) {  
+    clearInterval(timerInterval);
+    console.log("Time's up!");
+    alert("Time's up!");
+  }
+}, 1000);
+}
+
+
+
+// User Quiz Form Handling
+addQuestionForm.addEventListener("submit", (event) => {
+  event.preventDefault();
+  const questionText = document.querySelector("#question-text").value;
+  if (questionText) {
+    userQuiz.push({ question: questionText, options: [] });
+    document.querySelector("#question-text").value = "";
+    alert("Question added!");
+  }
+  const questionsList = document.querySelector("#questions-list");
+  const question = document.createElement("li");
+  question.innerText = questionText;
+  questionsList.appendChild(question);
+});
+
+
+
+
+// Add option for a user question
+addOptionForm.addEventListener("submit", (event) => {
+  event.preventDefault();
+  if (userQuiz.length === 0) {
+    alert("Please add a question first!");
+    return;
+  }
+  const optionText = document.querySelector("#option-text").value;
+  const isCorrect = document.querySelector("#is-correct").checked;
+  if (optionText) {
+    userQuiz[userQuiz.length - 1].options.push({ text: optionText, isCorrect });
+    document.querySelector("#option-text").value = "";
+    document.querySelector("#is-correct").checked = false;
+    alert("Option added!");
+  }
+});
+
+
+
+
+// Save user custom quiz
+saveQuizButton.addEventListener("click", () => {
+  if (userQuiz.length < 15) {
+    alert("You must add at least 15 questions to save the quiz.");
+    return;
+  }
+  alert("Your custom quiz has been saved successfully!");
+});
+
+
+
+
+
+// Start the quiz with the player names
+startQuizButton.addEventListener("click", () => {
+  playersInfo[0].name = players[0].value;
+  playersInfo[1].name = players[1].value;
+  player1.querySelector(".name").textContent = playersInfo[0].name;
+  player2.querySelector(".name").textContent = playersInfo[1].name;
+  updateScores();
+  navigateToPage(2);
+ // displayQuestion();
+  startTimer();
+});
+
+// Fetch the quiz questions from a json
+async function fetchQuizQuestions() {
+  try {
+    const response = await fetch("https://raw.githubusercontent.com/Aminata00/Aminata00.github.io-/refs/heads/main/APIS/data.json");
```

Comment:
The JSON you fetch here is this file: [Link](https://github.com/Aminata00/Aminata00.github.io-/blob/main/APIS/data.json#L2). The problem with it is that is missing the opening `{` character. That is why the JavaScript code cannot process it.